numpy/ma/core.py

```diff
@@ -6535,9 +6530,7 @@ def compressed(x):
         Equivalent method.
 
     """
-    if not isinstance(x, MaskedArray):
-        x = asanyarray(x)
-    return x.compressed()
+    return asanyarray(x).compressed()
```

Comment:
I agree. Until just now, I thought that this was only for methods that exist in `np`, but it is not